pax-utils: 1.3.8 -> 1.3.10

trofi opened the pull request:
Changes: https://gitweb.gentoo.org/proj/pax-utils.git/log/?h=v1.3.10


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

trofi committed:
pax-utils: 1.3.8 -> 1.3.10

Changes: https://gitweb.gentoo.org/proj/pax-utils.git/log/?h=v1.3.10

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/6214

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 464821`
Commit: [`6622dc0f4719f0fa321cf61e519bc2e130a89f32`](https://github.com/NixOS/nixpkgs/commit/6622dc0f4719f0fa321cf61e519bc2e130a89f32) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/6622dc0f4719f0fa321cf61e519bc2e130a89f32..pull/464821/head))
Merge: [`243a860d20a0a2cf2a3a9ab15d4936b5fd5f23a9`](https://github.com/NixOS/nixpkgs/commit/243a860d20a0a2cf2a3a9ab15d4936b5fd5f23a9)

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/20584047097

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-464821-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/0frlhzjbs02jnnhyhm6x3sw4kfscf4n3-pax-utils-1.3.10
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-464821-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/2911ybhgprfp0z3mqsrpl541y5ghv3pn-pax-utils-1.3.10
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>pax-utils</summary>
<pre>[20/33] Compiling C object dumpelf.fuzz.p/xfuncs.c.o
[21/33] Compiling C object dumpelf.fuzz.p/paxldso.c.o
[22/33] Compiling C object dumpelf.fuzz.p/paxelf.c.o
[23/33] Compiling C object dumpelf.fuzz.p/dumpelf.c.o
[24/33] Compiling C object fuzz-ar.p/paxinc.c.o
[25/33] Compiling C object fuzz-ar.p/security.c.o
[26/33] Compiling C object fuzz-ar.p/xfuncs.c.o
[27/33] Compiling C object fuzz-ar.p/fuzz-ar.c.o
FAILED: [code=1] fuzz-ar.p/fuzz-ar.c.o 
clang -Ifuzz-ar.p -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=gnu11 -DWANT_SECCOMP -DINCLUDE_GENERATE_VERSION_H &#x27;-DVERSION=&quot;1.3.10&quot;&#x27; -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -g3 -ggdb -fsanitize=fuzzer -fsanitize-coverage=edge -DPAX_UTILS_LIBFUZZ=1 -MD -MQ fuzz-ar.p/fuzz-ar.c.o -MF fuzz-ar.p/fuzz-ar.c.o.d -o fuzz-ar.p/fuzz-ar.c.o -c ../fuzz-ar.c
../fuzz-ar.c:18:7: error: call to undeclared function &#x27;memfd_create&#x27;; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   18 |         fd = memfd_create(&quot;fuzz-input.a&quot;, MFD_CLOEXEC);
      |              ^
../fuzz-ar.c:18:36: error: use of undeclared identifier &#x27;MFD_CLOEXEC&#x27;
   18 |         fd = memfd_create(&quot;fuzz-input.a&quot;, MFD_CLOEXEC);
      |                                           ^~~~~~~~~~~
2 errors generated.
[28/33] Linking target pspax
[29/33] Linking target scanelf
ninja: build stopped: subcommand failed.</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>pax-utils</summary>
<pre>[19/33] Compiling C object dumpelf.fuzz.p/xfuncs.c.o
[20/33] Compiling C object dumpelf.fuzz.p/paxldso.c.o
[21/33] Compiling C object scanelf.p/scanelf.c.o
[22/33] Compiling C object dumpelf.fuzz.p/paxelf.c.o
[23/33] Compiling C object fuzz-ar.p/security.c.o
[24/33] Compiling C object fuzz-ar.p/paxinc.c.o
[25/33] Compiling C object fuzz-ar.p/xfuncs.c.o
[26/33] Compiling C object fuzz-ar.p/fuzz-ar.c.o
FAILED: [code=1] fuzz-ar.p/fuzz-ar.c.o 
clang -Ifuzz-ar.p -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=gnu11 -DWANT_SECCOMP -DINCLUDE_GENERATE_VERSION_H &#x27;-DVERSION=&quot;1.3.10&quot;&#x27; -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -g3 -ggdb -fsanitize=fuzzer -fsanitize-coverage=edge -DPAX_UTILS_LIBFUZZ=1 -MD -MQ fuzz-ar.p/fuzz-ar.c.o -MF fuzz-ar.p/fuzz-ar.c.o.d -o fuzz-ar.p/fuzz-ar.c.o -c ../fuzz-ar.c
../fuzz-ar.c:18:7: error: call to undeclared function &#x27;memfd_create&#x27;; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   18 |         fd = memfd_create(&quot;fuzz-input.a&quot;, MFD_CLOEXEC);
      |              ^
../fuzz-ar.c:18:36: error: use of undeclared identifier &#x27;MFD_CLOEXEC&#x27;
   18 |         fd = memfd_create(&quot;fuzz-input.a&quot;, MFD_CLOEXEC);
      |                                           ^~~~~~~~~~~
2 errors generated.
[27/33] Compiling C object dumpelf.fuzz.p/dumpelf.c.o
[28/33] Linking target pspax
ninja: build stopped: subcommand failed.</pre>
</details>
</details>

trofi committed:
pax-utils: enable seccomp and fuzzong only on `linux`

On `darwin` the build fails on missing `memfd_create()` symbols as
https://hydra.nixos.org/build/314413362:

    ../fuzz-ar.c:18:7: error: call to undeclared function 'memfd_create'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
       18 |         fd = memfd_create("fuzz-input.a", MFD_CLOEXEC);
          |              ^

trofi commented:
I hope it should work on `darwin` now.

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 464821`
Commit: [`6a33ab8b96f052670051ffe87ffa1257bbaf58d4`](https://github.com/NixOS/nixpkgs/commit/6a33ab8b96f052670051ffe87ffa1257bbaf58d4) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/6a33ab8b96f052670051ffe87ffa1257bbaf58d4..pull/464821/head))
Merge: [`7d0fe677992c4f7f8f13f8132dc49f025d4f0bdb`](https://github.com/NixOS/nixpkgs/commit/7d0fe677992c4f7f8f13f8132dc49f025d4f0bdb)

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/20585092848

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-464821-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/3ld7szqcpc9p5w0p9mzp6s49i7adad0p-pax-utils-1.3.10
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-464821-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/vffjz5v0s2q9a1f5xhn5zm2qs29yfs2w-pax-utils-1.3.10
```
</details></li>
<li><details><summary><code>x86_64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-464821-x86_64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/cp028icxvhyh2kfas4lh3wvwczn8j0pc-pax-utils-1.3.10
```
</details></li>
<li><details><summary><code>aarch64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-464821-aarch64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://mdaniels5757-nix.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  mdaniels5757-nix.cachix.org-1:iT2D2XMbTS301DFoC8sTOSP86fXMPiGrvPnrEsQN5uo=
  ' \
  /nix/store/a0awg6hq81836570anr1v1xbb2qsq4l1-pax-utils-1.3.10
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pax-utils</li>
  </ul>
</details>

mdaniels5757 reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.
